kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusJobListener.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.services.prometheus;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.Summary;
+import org.jbpm.executor.AsynchronousJobEvent;
+import org.jbpm.executor.AsynchronousJobListener;
+import org.kie.api.executor.RequestInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.commons.lang3.StringUtils.defaultString;
+
+public class PrometheusJobListener implements AsynchronousJobListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PrometheusJobListener.class);
+
+    protected static final Gauge numberOfRunningJobs = Gauge.build()
+            .name("kie_server_job_running_total")
+            .help("Kie Server Running Jobs")
+            .labelNames("container_id", "command_name")
+            .register();
+    protected static final Counter numberOfJobsScheduled = Counter.build()
+            .name("kie_server_job_scheduled_total")
+            .help("Kie Server Started Jobs")
+            .labelNames("container_id", "command_name")
+            .register();
+    protected static final Counter numberOfJobsExecuted = Counter.build()
+            .name("kie_server_job_executed_total")
+            .help("Kie Server Executed Jobs")
+            .labelNames("container_id", "failed", "command_name")
+            .register();
+    protected static final Counter numberOfJobsCancelled = Counter.build()
+            .name("kie_server_job_cancelled_total")
+            .help("Kie Server Cancelled Jobs")
+            .labelNames("container_id", "command_name")
+            .register();
+    protected static final Summary jobDuration = Summary.build()
+            .name("kie_server_job_duration_seconds")
+            .help("Kie Server Job Duration")
+            .labelNames("container_id", "command_name")
+            .register();
+
+    private static ThreadLocal<Summary.Timer> threadLocalStartTime = new ThreadLocal<>();
```

Comment:
Does this one need to be static?

Reply:
not necessarily.

Reply:
If it does not have to be let's make it non-static. Thoughts @MarianMacik ?

Reply:
removed static reference.